File: sdk/node/src/types/mfa.ts
export interface MfaEnrollResponse {
  method_id: string;
  method_type: string;
  secret?: string;
  qr_code_uri?: string;
  backup_codes?: string[];
}

export interface SmsMfaEnrollResponse {
  method_id: string;
  method_type: string;
  phone_number: string;
  masked_phone: string;
  expires_at?: string;
}

export interface VerifyMfaRequest {
  code: string;
}

export interface EnrollSmsRequest {
  phone_number: string;
}

export interface EnrollWithTokenRequest {
  enrollment_token: string;
}

export interface VerifyWithTokenRequest {
  enrollment_token: string;
  code: string;
}

export interface MfaMethod {
  id: string;
  user_id: string;
  method_type: string;
  verified: boolean;
  name?: string;
  created_at?: string;
  last_used_at?: string;
}

File: sdk/python/coreauth/types/mfa.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class MfaEnrollResponse(BaseModel):
    method_id: Optional[str] = None
    method_type: Optional[str] = None
    secret: Optional[str] = None
    qr_code_uri: Optional[str] = None
    backup_codes: Optional[list[str]] = None


class SmsMfaEnrollResponse(BaseModel):
    method_id: Optional[str] = None
    method_type: Optional[str] = None
    phone_number: Optional[str] = None
    masked_phone: Optional[str] = None
    expires_at: Optional[str] = None


class VerifyMfaRequest(BaseModel):
    code: str


class EnrollSmsRequest(BaseModel):
    phone_number: str


class EnrollWithTokenRequest(BaseModel):
    enrollment_token: str


class VerifyWithTokenRequest(BaseModel):
    enrollment_token: str
    code: str


class MfaMethod(BaseModel):
    id: Optional[str] = None
    user_id: Optional[str] = None
    method_type: Optional[str] = None
    verified: Optional[bool] = None
    name: Optional[str] = None
    created_at: Optional[str] = None
    last_used_at: Optional[str] = None
